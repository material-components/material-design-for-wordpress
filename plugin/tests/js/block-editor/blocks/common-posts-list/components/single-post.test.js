/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import SinglePost from '../../../../../../assets/src/block-editor/blocks/common-posts-list/components/single-post';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <SinglePost { ...props } /> );
};

const baseProps = {
	style: 'masonry',
	attributes: {
		contentLayout: 'text-above-media',
		columns: 3,
		postsToShow: 12,
		outlined: false,
		displayPostDate: true,
		displayPostContent: true,
		postContentLength: 20,
		displayFeaturedImage: true,
		displayCommentsCount: true,
		displayPostAuthor: false,
		category: undefined,
	},
	post: {
		title: {
			rendered: 'This is a test post title #1',
		},
		excerpt: {
			rendered: 'This is a test post excerpt #1',
		},
		authorDisplayName: 'Test user #1',
		commentsCount: 1,
		date_gmt: ' 2020-02-25 08:20:38',
		featuredImageSourceUrl: 'http://example.com/test1.jpg',
	},
};

describe( 'SinglePost', () => {
	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'matches snapshot with style being "Masonry"', () => {
		const props = { ...baseProps };
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with style being "List"', () => {
		const props = { ...baseProps };
		props.style = 'list';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with style being "Grid"', () => {
		const props = { ...baseProps };
		props.style = 'grid';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
