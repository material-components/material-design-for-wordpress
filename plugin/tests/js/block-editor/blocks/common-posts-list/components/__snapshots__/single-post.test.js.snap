// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SinglePost matches snapshot with style being "Grid" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card single-post-card single-post-basic"
      >
        <div
          class="mdc-card__primary-action single-post-card__primary-action mdc-ripple-upgraded"
          tabindex="0"
        >
          <div
            class="single-post-card__primary"
          >
            <h2
              class="single-post-card__title mdc-typography mdc-typography--headline6"
            >
              <div>
                This is a test post title #1
              </div>
            </h2>
            <h3
              class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
            >
              <time
                datetime="2020-02-25 08:20:38"
              >
                February 25, 2020
              </time>
            </h3>
          </div>
          <div
            class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
            style="background-image: url(http://example.com/test1.jpg);"
          />
          <div
            class="single-post-card__secondary single-post-card__secondary-text-above-media mdc-typography mdc-typography--body2"
          >
            <div>
              This is a test post excerpt #1
            </div>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <button
              class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
            >
              <span
                class="mdc-button__ripple"
              />
              <i
                aria-hidden="true"
                class="material-icons mdc-button__icon"
              >
                comment
              </i>
              <span
                class="mdc-button__label"
              >
                1
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card single-post-card single-post-basic"
    >
      <div
        class="mdc-card__primary-action single-post-card__primary-action mdc-ripple-upgraded"
        tabindex="0"
      >
        <div
          class="single-post-card__primary"
        >
          <h2
            class="single-post-card__title mdc-typography mdc-typography--headline6"
          >
            <div>
              This is a test post title #1
            </div>
          </h2>
          <h3
            class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
          >
            <time
              datetime="2020-02-25 08:20:38"
            >
              February 25, 2020
            </time>
          </h3>
        </div>
        <div
          class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
          style="background-image: url(http://example.com/test1.jpg);"
        />
        <div
          class="single-post-card__secondary single-post-card__secondary-text-above-media mdc-typography mdc-typography--body2"
        >
          <div>
            This is a test post excerpt #1
          </div>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              1
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SinglePost matches snapshot with style being "List" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card single-post-card single-post-card__list single-post-basic"
      >
        <div
          class="mdc-card__primary-action single-post-card__primary-action"
          tabindex="0"
        >
          <div
            class="mdc-card__media mdc-card__media--square single-post-card__media"
            style="background-image: url(http://example.com/test1.jpg);"
          />
          <div
            class="single-post-card__primary"
          >
            <h2
              class="single-post-card__title mdc-typography mdc-typography--headline6"
            >
              <div>
                This is a test post title #1
              </div>
            </h2>
            <h3
              class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
            >
              <time
                datetime="2020-02-25 08:20:38"
              >
                February 25, 2020
              </time>
            </h3>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <button
              class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
            >
              <span
                class="mdc-button__ripple"
              />
              <i
                aria-hidden="true"
                class="material-icons mdc-button__icon"
              >
                comment
              </i>
              <span
                class="mdc-button__label"
              >
                1
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card single-post-card single-post-card__list single-post-basic"
    >
      <div
        class="mdc-card__primary-action single-post-card__primary-action"
        tabindex="0"
      >
        <div
          class="mdc-card__media mdc-card__media--square single-post-card__media"
          style="background-image: url(http://example.com/test1.jpg);"
        />
        <div
          class="single-post-card__primary"
        >
          <h2
            class="single-post-card__title mdc-typography mdc-typography--headline6"
          >
            <div>
              This is a test post title #1
            </div>
          </h2>
          <h3
            class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
          >
            <time
              datetime="2020-02-25 08:20:38"
            >
              February 25, 2020
            </time>
          </h3>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              1
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SinglePost matches snapshot with style being "Masonry" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card single-post-card single-post-basic"
      >
        <div
          class="mdc-card__primary-action single-post-card__primary-action mdc-ripple-upgraded"
          tabindex="0"
        >
          <div
            class="single-post-card__primary"
          >
            <h2
              class="single-post-card__title mdc-typography mdc-typography--headline6"
            >
              <div>
                This is a test post title #1
              </div>
            </h2>
            <h3
              class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
            >
              <time
                datetime="2020-02-25 08:20:38"
              >
                February 25, 2020
              </time>
            </h3>
          </div>
          <div
            class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
            style="background-image: url(http://example.com/test1.jpg);"
          />
          <div
            class="single-post-card__secondary single-post-card__secondary-text-above-media mdc-typography mdc-typography--body2"
          >
            <div>
              This is a test post excerpt #1
            </div>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <button
              class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
            >
              <span
                class="mdc-button__ripple"
              />
              <i
                aria-hidden="true"
                class="material-icons mdc-button__icon"
              >
                comment
              </i>
              <span
                class="mdc-button__label"
              >
                1
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card single-post-card single-post-basic"
    >
      <div
        class="mdc-card__primary-action single-post-card__primary-action mdc-ripple-upgraded"
        tabindex="0"
      >
        <div
          class="single-post-card__primary"
        >
          <h2
            class="single-post-card__title mdc-typography mdc-typography--headline6"
          >
            <div>
              This is a test post title #1
            </div>
          </h2>
          <h3
            class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
          >
            <time
              datetime="2020-02-25 08:20:38"
            >
              February 25, 2020
            </time>
          </h3>
        </div>
        <div
          class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
          style="background-image: url(http://example.com/test1.jpg);"
        />
        <div
          class="single-post-card__secondary single-post-card__secondary-text-above-media mdc-typography mdc-typography--body2"
        >
          <div>
            This is a test post excerpt #1
          </div>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              1
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
